solutions/baltic-20-VillageMinimum.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: baltic-20-VillageMinimum
+source: Baltic OI 2020
+title: Village (Minimum)
+author: Chuyang Wang 
+---
+
+[Official Analysis](http://www.boi2020.lv/data/spoilers/village.pdf)
+
+
+## Explanation
+
+This problem can be solved by a greedy approach. In particular, we perform a
+depth-first search on the given tree. For each node (villager) that is still at
+its original position, we swap it with its only neighbor, namely its parent
+node. For the root node which does not have a parent node, we can swap it with
+any of its children.
+
+By swapping two nodes where at least one of them is not processed yet, we can
+guarantee that no node will be swapped back to its original position. We can
+also show that swapping between neighbors is the optimal solution: Suppose we
+would swap two unprocessed nodes which are not neighbors. The distance the two
+nodes need to travel would then be equal or larger than 4. If we instead just
```

Comment:
4 seems like a really specific number, could you explain this w/ an example maybe?

Reply:
i'd add an illustration please